mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRThumbnailImageImpl.java

```diff
@@ -66,15 +67,33 @@ protected MCRTileInfo createTileInfo(String id) throws MCRIIIFImageNotFoundExcep
                         .anyMatch(type::equals)) {
                         final String maindoc = derLink.getMainDoc();
                         if (maindoc != null) {
-                            final MCRTileInfo mcrTileInfo = new MCRTileInfo(derLink.getXLinkHref(), maindoc, null);
-                            final Optional<Path> tileFile = this.tileFileProvider.getTileFile(mcrTileInfo);
-                            if (tileFile.isPresent() && Files.exists(tileFile.get())) {
-                                return mcrTileInfo;
+                            Optional<MCRTileInfo> tileInfoForFile = createTileInfoForFile(derLink.getXLinkHref(),
+                                maindoc);
+                            if (tileInfoForFile.isPresent()) {
+                                return tileInfoForFile.get();
                             }
                         }
                     }
                 }
             }
+            Optional<String> linkedImage = MCRLinkTableManager.instance()
+                .getDestinationOf(mcrID, MCRLinkTableManager.ENTRY_TYPE_DERIVATE_LINK).stream().findFirst();
+            if (linkedImage.isPresent()) {
+                MCRObjectID deriID = MCRObjectID.getInstance(
+                    linkedImage.get().substring(0, linkedImage.get().indexOf("/")));
+                MCRDerivate deri = MCRMetadataManager.retrieveMCRDerivate(deriID);
+                if (deri.getDerivate().getClassifications().stream()
+                    .map(classi -> classi.getClassId() + ":" + classi.getCategId())
```

Comment:
This can be MCRCategoryID::toString

Reply:
deri.getDerivate().getClassifications() returns a ArrayList<MCRMetaClassification> and MCRMetaClassification::toString does not return the right format.

Reply:
May be we should a the following methods to `MCRMetaClassification`:
```
public final MCRCategoryID getCategoryID(){
        return category;
}

public final void setValue(MCRCategoryID category) throws MCRException {
        this.category = category;
    }
```
In my opinion it is useful to have the MCRCategoryID object available.

Reply:
Should I add this in this Pull request?

Reply:
@sebhofmann  and I came to the conclusion to leave it as it is for now